examples/window-app/efr32/BUILD.gn

```diff
@@ -91,14 +92,28 @@ if (chip_enable_wifi) {
   }
   wifi_sdk_dir = "${chip_root}/src/platform/EFR32/wifi"
   efr32_lwip_defs = [ "LWIP_NETIF_API=1" ]
-  efr32_lwip_defs += [
-    "LWIP_IPV4=1",
-    "LWIP_ARP=1",
-    "LWIP_ICMP=1",
-    "LWIP_DHCP=1",
-    "LWIP_IPV6_ND=1",
-    "LWIP_IGMP=1",
-  ]
+  if (lwip_ipv4) {
+    efr32_lwip_defs += [
+      "LWIP_IPV4=1",
+
+      # adds following options to provide
+      # them to .cpp source files
+      # flags ported from lwipopts file
+      # TODO: move lwipopts to one location
+      "LWIP_ARP=1",
+      "LWIP_ICMP=1",
+      "LWIP_IGMP=1",
+      "LWIP_DHCP=1",
+      "LWIP_DNS=0",
+    ]
+  } else {
+    efr32_lwip_defs += [ "LWIP_IPV4=0" ]
+  }
```

Comment:
Can't we factorize all of this ? 
This is the same pattern all over the demos

Something in this spirit
import("${chip_root}/src/platform/EFR32/lwip_args.gni

Reply:
That is a good amount of work so we can merge this one, and then do that factorizing work and submit another IOTREQ if that's OK.

Reply:
no i kinda disagree this doesn't not look a too big effort , looking at these files , this create a big diff all over and then you will have to submit another PR and remove everything you just added in this one... 
rebase etc... that gonna take forever. I think it would be better to do it right away ! but i don't want to be a blocker , so i am gonna approve.